src/components/structures/RoomView.js

```diff
@@ -995,8 +995,18 @@ module.exports = React.createClass({
 
             var eventId = mxEv.getId();
             var highlight = (eventId == this.props.highlightedEventId);
+
+            var scrollToken = eventId;
+            // we can't use local echoes as scroll tokens, because their event
+            // IDs change.
+            if (mxEv.status) {
+                scrollToken = undefined;
+            }
```

Comment:
Would probably be neater written as:
```javascript
// we can't use local echoes as scroll tokens, because their event IDs change.
// Local echos have a send "status".
var scrollToken = mxEv.status ? undefined : eventId;
```